zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/NetworkUtil.java

```diff
@@ -120,7 +120,7 @@ public static String getLocalhostName() {
         try {
             return InetAddress.getLocalHost().getHostName();
         } catch (final UnknownHostException e) {
-            throw new RuntimeException("unable to retrieve localhost name");
+            return "UNKNOWN";
```

Comment:
Why? If you return "UNKNOWN" when getLocalhostName() gets an error, the application will continue executing and cause bigger trouble.

Reply:
This static API is not used anywhere else. I don't think it would cause rebalance errors.

Reply:
Then I think we should not change it.
1. Throw exception is the right way when the tool hits any problem.
2. If you are convinced that no one is using this method, just remove or deprecate it.

This change does not make sense to me.